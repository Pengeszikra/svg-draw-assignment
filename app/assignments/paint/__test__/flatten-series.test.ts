import Flatten from '@flatten-js/core'; 

test ('explore Flatten import', () => {
  expect (
    Flatten
  ).toMatchSnapshot();
});

test ('line intersection', () => {
  const {segment} = Flatten;
  let s1 = segment(10,10,200,200);
  let s2 = segment(10,160,200,30);
  
  let ip = s1.intersect(s2);
  

  expect (
    ip
  ).toMatchSnapshot();
});


