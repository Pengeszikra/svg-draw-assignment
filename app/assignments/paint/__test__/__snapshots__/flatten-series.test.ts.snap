// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`explore Flatten import 1`] = `
Object {
  "Arc": [Function],
  "BOUNDARY": 2,
  "BooleanOperations": Object {
    "BOOLEAN_INTERSECT": 2,
    "BOOLEAN_SUBTRACT": 3,
    "BOOLEAN_UNION": 1,
    "calculateIntersections": [Function],
    "innerClip": [Function],
    "intersect": [Function],
    "outerClip": [Function],
    "removeNotRelevantChains": [Function],
    "removeOldFaces": [Function],
    "restoreFaces": [Function],
    "subtract": [Function],
    "unify": [Function],
  },
  "Box": [Function],
  "CCW": true,
  "CONTAINS": 3,
  "CW": false,
  "Circle": [Function],
  "Distance": [Function],
  "END_VERTEX": 2,
  "Edge": [Function],
  "Errors": [Function],
  "Face": [Function],
  "INSIDE": 1,
  "INTERLACE": 4,
  "Inversion": [Function],
  "Line": [Function],
  "Matrix": [Function],
  "Multiline": [Function],
  "NOT_VERTEX": 0,
  "ORIENTATION": Object {
    "CCW": -1,
    "CW": 1,
    "NOT_ORIENTABLE": 0,
  },
  "OUTSIDE": 0,
  "OVERLAP_OPPOSITE": 2,
  "OVERLAP_SAME": 1,
  "PIx2": 6.283185307179586,
  "PlanarSet": [Function],
  "Planar_set": undefined,
  "Point": [Function],
  "Polygon": [Function],
  "Ray": [Function],
  "Ray_shooting": undefined,
  "Relations": Object {
    "contain": [Function],
    "cover": [Function],
    "covered": [Function],
    "disjoint": [Function],
    "equal": [Function],
    "inside": [Function],
    "intersect": [Function],
    "relate": [Function],
    "touch": [Function],
  },
  "START_VERTEX": 1,
  "Segment": [Function],
  "Utils": Object {
    "DECIMALS": 3,
    "EQ": [Function],
    "EQ_0": [Function],
    "GE": [Function],
    "GT": [Function],
    "LE": [Function],
    "LT": [Function],
    "getTolerance": [Function],
    "setTolerance": [Function],
  },
  "Vector": [Function],
  "arc": [Function],
  "box": [Function],
  "circle": [Function],
  "inversion": [Function],
  "line": [Function],
  "matrix": [Function],
  "multiline": [Function],
  "point": [Function],
  "polygon": [Function],
  "ray": [Function],
  "segment": [Function],
  "vector": [Function],
}
`;

exports[`line intersection 1`] = `
Array [
  Object {
    "name": "point",
    "x": 99.06250000000001,
    "y": 99.06250000000001,
  },
]
`;
